android/src/main/java/com/gaspardbruno/staticsafeareainsets/RNStaticSafeAreaInsetsModule.java

```diff
@@ -28,23 +35,59 @@ public String getName() {
   @Override
   public Map<String, Object> getConstants() {
     final Map<String, Object> constants = new HashMap<>();
-
-    if (getCurrentActivity() != null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      final Activity activity = getCurrentActivity();
+    // Window insets are parts of the window that are covered by system views (status bar,
+    // navigation bar, notches). There are no apis the get these values for android < M so we
+    // do a best effort polyfill.
+    EdgeInsets windowInsets = new EdgeInsets(0, 0, 0, 0);
+    final Activity activity = getCurrentActivity();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
       final View view = activity.getWindow().getDecorView();
       final WindowInsets insets = view.getRootWindowInsets();
+      windowInsets = new EdgeInsets(
+          insets.getSystemWindowInsetTop(),
+          insets.getSystemWindowInsetRight(),
+          insets.getSystemWindowInsetBottom(),
+          insets.getSystemWindowInsetLeft());
+    }
+    final Resources resources = mReactContext.getResources();
+    final int rotation = mWindowManager.getDefaultDisplay().getRotation();
+    float statusBarHeight = 0;
+    int resourceId = resources.getIdentifier("status_bar_height", "dimen", "android");
+    if (resourceId > 0) {
+      statusBarHeight = PixelUtil.toDIPFromPixel(resources.getDimensionPixelSize(resourceId));
+    }
+    float navbarHeight = 0;
+    resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android");
+    if (resourceId > 0) {
+      navbarHeight = PixelUtil.toDIPFromPixel(resources.getDimensionPixelSize(resourceId));
+    }
+    // Fix for views drawing under the navbar & statusbar
+    windowInsets = new EdgeInsets(
+        windowInsets.top != 0 ? windowInsets.top : statusBarHeight,
+        windowInsets.right != 0 ? windowInsets.right : rotation == Surface.ROTATION_90 ? navbarHeight : 0,
+        windowInsets.bottom != 0 ? windowInsets.bottom : rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180 ? navbarHeight : 0,
+        windowInsets.left != 0 ? windowInsets.left : rotation == Surface.ROTATION_270 ? navbarHeight : 0);
+
 
-      constants.put("safeAreaInsetsTop", insets.getSystemWindowInsetTop());
-      constants.put("safeAreaInsetsBottom", insets.getSystemWindowInsetBottom());
-      constants.put("safeAreaInsetsLeft", insets.getSystemWindowInsetLeft());
-      constants.put("safeAreaInsetsRight", insets.getSystemWindowInsetRight());
-    } else {
-      constants.put("safeAreaInsetsTop", 0);
-      constants.put("safeAreaInsetsBottom", 0);
-      constants.put("safeAreaInsetsLeft", 0);
-      constants.put("safeAreaInsetsRight", 0);
+    // Calculate the part of the root view that overlaps with window insets.
+    if (activity != null) {
```

Comment:
Same as before, why are you adding this `RootView` logic here?

Reply:
`RootView` is used for devices, such as the Google Pixel XL, that have a **notch** or any other screen decoration. We use the `windowWidth` and `windowHeight` got from the `RootView` to adjust the offset created by those decorations.